### PR TITLE
Updates types to match Photoshop version 24.3

### PR DESCRIPTION
**Pull request by jardicc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[Photoshop docs](https://developer.adobe.com/photoshop/uxp/2022/ps_reference/classes/photoshop/)>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by heewooa — APPROVED** [review]

**Review by heewooa — APPROVED** [review]

File: types/photoshop/index.d.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
-// Type definitions for Photoshop JavaScript API 23.0
+// Type definitions for Photoshop JavaScript API 24.3
 // Project: https://adobe.io/photoshop/uxp
 // Definitions by:  Adobe Photoshop <https://github.com/adobe-photoshop>
 //                  Barkin Aygun <https://github.com/baaygun>
 //                  Heewoo Ahn <https://github.com/heewooa>
+//                  Jaroslav Bereza <https://github.com/jardicc>
```
Comment (jakebailey) [inline types/photoshop/index.d.ts:6]: Note that this is only used for managing the review lifecycle of DT PRs, and adding yourself here will ping you for all changes to this package and let you review it.

It's fine if that's what you want (and the other maintainers agree), but if you're just looking for credit, that is handled by the git history.

Comment (jardicc) [reply types/photoshop/index.d.ts:6]: Hello, this is intentional. I created parts of that DOM myself and typings were autogenerated from the source code. In future we will likely move typings elsewhere but before we decide we want to update it here.

Also we generate documentation from source code and that has some requirements regarding JSDOC and function signatures.

Comment (jakebailey) [reply types/photoshop/index.d.ts:6]: Thanks, just checking!

**Review by jakebailey — APPROVED** [review]

**Comment by jardicc** [comment]
Ready to merge

**Pull request merged**